### DISCUSSION TITLE
Issue Implementing Forecaster with Deep Learning Tutorial

### DISCUSSION THREAD

**Discussion by bassamabdelghani** [JoaquinAmatRodrigo/skforecast#756] [Q&A] [answered]
Hello,

I attempted to implement the Forecaster with Deep Learning tutorial included in the library. I used the same dataset and followed the steps outlined in the tutorial. However, I encountered an error when calling the ForecasterRnn at the following step:

# Forecaster Creation
# ==============================================================================
forecaster = ForecasterRnn(
    regressor=model,
    levels=levels,
    lags=lags,
    transformer_series=MinMaxScaler(),
    fit_kwargs={
        "epochs": 10,  # Number of epochs to train the model.
        "batch_size": 32,  # Batch size to train the model.
        "callbacks": [
            EarlyStopping(monitor="val_loss", patience=5)
        ],  # Callback to stop training when it is no longer learning.
        "series_val": data_val,  # Validation data for model training.
    },
)    
Upon execution, I receive the following error:

AttributeError: 'Reshape' object has no attribute 'output_shape'

During handling of the above exception, another exception occurred:

TypeError: Input shape of the regressor should be Input(shape=(lags, n_series)).
It appears the issue is related to the input shape of the regressor. This might be due to compatibility issues between different library versions (Keras and TensorFlow).

Here are the versions of the libraries I am using:

keras               3.4.1
matplotlib          3.8.4
numpy               1.26.4
pandas              2.2.2
plotly              5.22.0
session_info        1.0.0
skforecast          0.12.1
sklearn             1.4.2
tensorflow          2.16.0-rc0
IPython             8.25.0
jupyter_client      8.6.0
jupyter_core        5.7.2
jupyterlab          4.0.11
notebook            7.0.8
Python              3.12.4 | packaged by Anaconda, Inc. | (main, Jun 18 2024, 15:03:56) [MSC v.1929 64 bit (AMD64)]
Windows-11-10.0.22631-SP0

Could you please provide guidance on how to resolve this issue?

Thank you.

Best regards,

Bassam

**Comment by JavierEscobarOrtiz** [comment]
Hi @bassamabdelghani, maybe @FernandoCarazoMelo who is the creator of the RNN module, can help you 😄

**Comment by FernandoCarazoMelo** [comment]
Hi @bassamabdelghani ,

The issue you're encountering is related to version incompatibility. As described in the [docu](https://skforecast.org/0.12.1/quick-start/how-to-install), the current implementation of skforecast requires tensorflow>=2.13, <2.16. You are using the 2.16 version. 

In the next release of skforecast we will add compatibility with tensorflow 2.16 & keras 3

Let me know if you have any other issues.

Feature #727

**Reply by bassamabdelghani** [reply]
> Hi @bassamabdelghani ,
> 
> The issue you're encountering is related to version incompatibility. As described in the [docu](https://skforecast.org/0.12.1/quick-start/how-to-install), the current implementation of skforecast requires tensorflow>=2.13, <2.16. You are using the 2.16 version.
> 
> In the next release of skforecast we will add compatibility with tensorflow 2.16 & keras 3
> 
> Let me know if you have any other issues.
> 
> Feature #727

i expected that, will try to downgrade them , do you know when it is expected to release the updated version from skforcast?

**Comment by JavierEscobarOrtiz** [comment, marked as answer]
Hi,

The `ForecasterRnn` and the `create_and_compile_model` function have been refactored in `skforecast 0.17.0`. They now support Keras 3 and tensorflow 2.19.0.

https://skforecast.org/0.17.0/user_guides/forecasting-with-deep-learning-rnn-lstm

Thank you,

Javi
